src/test/ui/borrowck/borrowck-borrow-from-temporary.nll.stderr

```diff
@@ -1,18 +1,11 @@
-error[E0716]: temporary value dropped while borrowed
-  --> $DIR/borrowck-borrow-from-temporary.rs:19:24
+error[E0515]: cannot return value referencing temporary value
+  --> $DIR/borrowck-borrow-from-temporary.rs:20:5
```

Comment:
Do we want to fix this in this PR? I think we had chosen to use `E0716` specifically so that we could give targeted help text (cc @pnkfelix) but I'm also ok with landing this PR as is I think.

Reply:
this is fine with me. i'd be more concerned if this were a case that removed the extra notes that I added in #54782, but that does not appear to be the case here.